### ISSUE TITLE
Don't supress interceptors errors

### ISSUE THREAD

**Issue by mikicho** [mswjs/interceptors#452]
Currently, we [suppress all errors for specific types](https://github.com/mswjs/interceptors/blob/v0.25.6/src/interceptors/ClientRequest/NodeClientRequest.ts#L357), which prevent the user to throw these error within the interceptors handler:
```js
const { ClientRequestInterceptor } = require('@mswjs/interceptors/ClientRequest')
const http = require('http');
const { inherits } = require('util');

const interceptor = new ClientRequestInterceptor({
  name: 'my-interceptor',
})
interceptor.apply();
interceptor.on('request', ({request}) => {
  throw new NetConnectNotAllowedError('example.test', '/')
});

http.get('http://nowhere.com/', res => {
  res.on('data', () => {})
  res.on('end', () => {
    console.log(1) 
  })
})
.on('error', e => {
  console.log('custom error') // never get called!!!
})

function NetConnectNotAllowedError(host, path) {
  Error.call(this)

  this.name = 'NetConnectNotAllowedError'
  this.code = 'ENETUNREACH'
  this.message = `Nock: Disallowed net connect for "${host}${path}"`

  Error.captureStackTrace(this, this.constructor)
}

inherits(NetConnectNotAllowedError, Error)
```

As I see it, we can do one of two things:

1. have a flag that indicates that the error is from the user and shouldn't be suppressed.
2. Set the default `resourceSource` to `pass` or to `undefined`

**Comment by kettanaito** [comment]
Oh, I see. Since the thrown errors result in `error` being emitted, and we have the suppressing logic in there, those user-defined errors never get emitted.

I think this should be fairly easy to fix by introducing a state that tells the interceptor that we are past socket connection phase, and it should no longer suppress the errors. This is a bit naive and relies on the suppressed errors originating from the connection alone but that's also the only use case we have for those suppressed errors as of now.

**Comment by mikicho** [comment]
Great. I can open a PR if needed.

**Comment by kettanaito** [comment]
That would be much appreciated! Let's iterate on it together.

**Comment by kettanaito** [comment]
## Released: v0.25.7 🎉

This has been released in v0.25.7!

- 📄 [**Release notes**](https://github.com/mswjs/interceptors/releases/tag/v0.25.7)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/interceptors/v/0.25.7)

Make sure to always update to the latest version (`npm i @mswjs/interceptors@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
